storage/schemadb/src/lib.rs

```diff
@@ -224,12 +224,17 @@ impl DB {
     pub fn open_readonly<P: AsRef<Path>>(
         path: P,
         cf_opts_map: ColumnFamilyOptionsMap,
-        db_log_dir: &str,
+        db_log_dir: P,
     ) -> Result<Self> {
         let mut db_opts = DBOptions::new();
 
         db_opts.create_if_missing(false);
-        db_opts.set_db_log_dir(db_log_dir);
+        db_opts.set_db_log_dir(db_log_dir.as_ref().to_str().ok_or_else(|| {
+            format_err!(
+                "db_log_dir {:?} can not be converted to string.",
+                db_log_dir.as_ref()
+            )
+        })?);
```

Comment:
```suggestion
        if !db_exists(path.as_ref()) {
            bail!("xx")
        }
```

Reply:
I was blind. ignore. 

use `db_exist` to check the `path` though, and delete the check outside this.